core/modules/SConscript

```diff
@@ -163,6 +181,18 @@ def standardModule(env, exclude=None, module=None, test_libs=None, unit_tests=No
     py_installed = env.Install("$prefix/bin", py_scripts)
     build_data['install'] += py_installed
 
+    # inject binary targets built of the optionally provided C/C++ files
+    if bin_cc_targets is not None:
+        for f,dependencies in bin_cc_targets.items():
+            source = os.path.basename(str(f))
+            target = source[:-3]
+            p = env.Program(
+                target=target,
+                source=[source],
+                LIBS=dependencies,
+                LIBPATH=['$build_dir'] + env['LIBPATH'])
```

Comment:
Looks a bit  complicated, I'd think this should work:

    for f, dependencies in bin_cc_targets.items():
        p = env.Program(f, LIBS=dependencies, LIBPATH=['$build_dir'] + env['LIBPATH'])

Reply:
I will give it a try, and if it works then I'll make mods to the code.